Build fails with

```
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/qizipyz9y17nr4w4gmxvwd3x4k0bp2rh-libxcrypt-4.4.38/include -fPIC -I/nix/store/2mab9iiwhcqwk75qwvp3zv0bvbiaq6cs-python3-3.13.3/include/python3.13 -c efl/eo/efl.eo.c -o build/temp.linux-x86_64-cpython-313/efl/eo/efl.eo.o -pthread -I/nix/store/n0ss07gpw5ih7pda88y8s1nncnpkzhgk-efl-1.28.1/include/eo-1 -I/nix/store/n0ss07gpw5ih7pda88y8s1nncnpkzhgk-efl-1.28.1/include/eina-1 -I/nix/store/n0ss07gpw5ih7pda88y8s1nncnpkzhgk-efl-1.28.1/include/eina-1/eina -fno-var-tracking-assignments -Wno-misleading-indentation -Wno-deprecated-declarations -Wno-unused-variable -Wno-format-security
efl/eo/efl.eo.c: In function ‘__pyx_tp_dealloc_3efl_2eo_Eo’:
efl/eo/efl.eo.c:5443:104: error: implicit declaration of function ‘_PyGC_FINALIZED’ []
 5443 |   if (unlikely(PyType_HasFeature(Py_TYPE(o), Py_TPFLAGS_HAVE_FINALIZE) && Py_TYPE(o)->tp_finalize) && !_PyGC_FINALIZED(o)) {
      |                                                                                                        ^~~~~~~~~~~~~~~
efl/eo/efl.eo.c: In function ‘__Pyx_PyInt_As_int’:
efl/eo/efl.eo.c:9221:27: error: too few arguments to function ‘_PyLong_AsByteArray’
 9221 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,
      |                           ^~~~~~~~~~~~~~~~~~~
In file included from /nix/store/2mab9iiwhcqwk75qwvp3zv0bvbiaq6cs-python3-3.13.3/include/python3.13/longobject.h:107,
                 from /nix/store/2mab9iiwhcqwk75qwvp3zv0bvbiaq6cs-python3-3.13.3/include/python3.13/Python.h:81,
                 from efl/eo/efl.eo.c:38:
/nix/store/2mab9iiwhcqwk75qwvp3zv0bvbiaq6cs-python3-3.13.3/include/python3.13/cpython/longobject.h:111:17: note: declared here
  111 | PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
      |                 ^~~~~~~~~~~~~~~~~~~
efl/eo/efl.eo.c: In function ‘__Pyx_PyInt_As_long’:
efl/eo/efl.eo.c:9493:27: error: too few arguments to function ‘_PyLong_AsByteArray’
 9493 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,
      |                           ^~~~~~~~~~~~~~~~~~~
/nix/store/2mab9iiwhcqwk75qwvp3zv0bvbiaq6cs-python3-3.13.3/include/python3.13/cpython/longobject.h:111:17: note: declared here
  111 | PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
      |                 ^~~~~~~~~~~~~~~~~~~
error: command '/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/gcc' failed with exit code 1
ERROR Backend subprocess exited when trying to invoke build_wheel
```

Probably due to the last Python 3.13 update.

--- a/pkgs/development/python-modules/python-efl/default.nix
+++ b/pkgs/development/python-modules/python-efl/default.nix
@@ -13,6 +13,8 @@
 
   dbus-python,
 
+  pytestCheckHook,
+
   directoryListingUpdater,
 }:
 
@@ -48,7 +50,30 @@ buildPythonPackage rec {
     NIX_CFLAGS_COMPILE="$(pkg-config --cflags efl evas) $NIX_CFLAGS_COMPILE"
   '';
 
-  doCheck = false;
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  preCheck = ''
+    # make sure we load the library from $out instead of the cwd
+    # because cwd doesn't contain the built extensions
+    rm -r efl/
+
+    patchShebangs tests/ecore/exe_helper.sh
+
+    # use the new name instead of the removed alias
+    substituteInPlace tests/evas/test_01_rect.py \
+      --replace-fail ".assert_(" ".assertTrue("
+  '';
+
+  pytestFlagsArray = [ "tests/" ];
+
+  disabledTestPaths = [
+    "tests/dbus/test_01_basics.py" # needs dbus daemon
+    "tests/ecore/test_09_file_download.py" # uses network
+    "tests/ecore/test_11_con.py" # uses network
+    "tests/elementary/test_02_image_icon.py" # RuntimeWarning: Setting standard icon failed
+  ];
 
   passthru.updateScript = directoryListingUpdater { };
 

--- a/pkgs/development/python-modules/python-efl/default.nix
+++ b/pkgs/development/python-modules/python-efl/default.nix
@@ -2,11 +2,17 @@
   lib,
   fetchurl,
   buildPythonPackage,
+  pythonAtLeast,
+
   pkg-config,
-  python,
-  dbus-python,
-  packaging,
+
   enlightenment,
+
+  packaging,
+  setuptools,
+
+  dbus-python,
+
   directoryListingUpdater,
 }:
 
@@ -15,10 +21,13 @@
 buildPythonPackage rec {
   pname = "python-efl";
   version = "1.26.1";
-  format = "setuptools";
+  pyproject = true;
+
+  # As of 1.26.1, native extensions fail to build with python 3.13+
+  disabled = pythonAtLeast "3.13";
 
   src = fetchurl {
-    url = "http://download.enlightenment.org/rel/bindings/python/${pname}-${version}.tar.xz";
+    url = "http://download.enlightenment.org/rel/bindings/python/python-efl-${version}.tar.xz";
     hash = "sha256-3Ns5fhIHihnpDYDnxvPP00WIZL/o1UWLzgNott4GKNc=";
   };
 
@@ -26,21 +35,17 @@ buildPythonPackage rec {
 
   buildInputs = [ enlightenment.efl ];
 
-  propagatedBuildInputs = [
-    dbus-python
+  build-system = [
     packaging
+    setuptools
+  ];
+
+  dependencies = [
+    dbus-python
   ];
 
   preConfigure = ''
     NIX_CFLAGS_COMPILE="$(pkg-config --cflags efl evas) $NIX_CFLAGS_COMPILE"
-  '';
-
-  preBuild = ''
-    ${python.pythonOnBuildForHost.interpreter} setup.py build_ext
-  '';
-
-  installPhase = ''
-    ${python.pythonOnBuildForHost.interpreter} setup.py install --prefix=$out --single-version-externally-managed
   '';
 
   doCheck = false;

